marshal_test.go

```diff
@@ -302,15 +304,41 @@ var marshalTests = []struct {
 		F1: "test",
 	},
 	expectURLString: "http://localhost?a=b&f1=test",
-},
-	{
-		about:           "url with query parameters no form",
-		urlString:       "http://localhost?a=b",
-		method:          "POST",
-		val:             &struct{}{},
-		expectURLString: "http://localhost?a=b",
+}, {
+	about:           "url with query parameters no form",
+	urlString:       "http://localhost?a=b",
+	method:          "POST",
+	val:             &struct{}{},
+	expectURLString: "http://localhost?a=b",
+}, {
+	about:     "struct with headers",
+	urlString: "http://localhost:8081/:F1",
+	val: &struct {
+		F1 int    `httprequest:",path"`
+		F2 string `httprequest:",form"`
+		F3 string `httprequest:",header"`
+	}{
+		F1: 99,
+		F2: "some text",
+		F3: "some more text",
 	},
-}
+	expectURLString: "http://localhost:8081/99?F2=some+text",
+	expectHeader:    http.Header{"F3": []string{"some more text"}},
+}, {
+	about:     "struct with header slice",
+	urlString: "http://localhost:8081/:F1",
+	val: &struct {
+		F1 int      `httprequest:",path"`
+		F2 string   `httprequest:",form"`
+		F3 []string `httprequest:",header"`
+	}{
+		F1: 99,
+		F2: "some text",
+		F3: []string{"A", "B", "C"},
+	},
+	expectURLString: "http://localhost:8081/99?F2=some+text",
+	expectHeader:    http.Header{"F3": []string{"A", "B", "C"}},
+}}
```

Comment:
What about a test marshaling a header into some other type like an integer?

And perhaps a test with a value that implements both HeaderSetter and has custom headers (which wins?)